### ISSUE TITLE
z-score issue

### ISSUE THREAD

**Issue by pavanramkumar** [pavanramkumar/pyglmnet#7]
In the demo and example, z-score the training set and then use the mean and std of training set to z-score test set. This is best practice for streaming test set data

**Comment by jasmainak** [comment]
I think you can use `StandardScaler` in a scikit-learn `Pipeline` object. Something like:

```
from sklearn.pipeline import Pipeline
from sklearn import cross_val_score
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler
glm = GLM()
clf = Pipeline([("scaler", scaler), ("glm", glm)])
score = cross_val_score(X, y)
```
for this `GLM` should inherit from `BaseEstimator` in scikit-learn (and hopefully it will work -- fingers crossed). The advantage of this approach is that you don't reinvent the wheel with respect to cross validation, grid search and preprocessing -- goodies that scikit-learn already has.

**Comment by hugoguh** [comment]
I replaced `zscore` with `StandardScaler`.
I think maybe `Pipeline` might be too much for the `example notebook`?

**Comment by jasmainak** [comment]
yeah maybe. I vote for simplicity :)

**Comment by hugoguh** [comment]
working on this. also removing standardization from inside the function. it’s doing zscoring in predict and not in fit. shouldn’t do it in either

**Comment by pavanramkumar** [comment]
zscore is now gone from pyglmnet. we can replace zscore with StandardScaler in the demo if needed.

**Comment by hugoguh** [comment]
I’ll do it now

**Comment by pavanramkumar** [comment]
this is solved by #32

**Comment by jasmainak** [comment]
you can put the magic word "closes #7" in the PR description so that the issue automatically closes when the PR is merged :)

**Comment by titipata** [comment]
Only @pavanramkumar can do it.
